pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -517,55 +496,28 @@ func (c *Reconciler) createBuildPod(tr *v1alpha1.TaskRun, ts *v1alpha1.TaskSpec,
 // an entrypoint cache creates a build where all entrypoints are switched to
 // be the entrypoint redirector binary. This function assumes that it receives
 // its own copy of the BuildSpec and modifies it freely
-func createRedirectedBuild(kubeclient kubernetes.Interface, build *buildv1alpha1.Build, tr *v1alpha1.TaskRun, cache *entrypoint.Cache, logger *zap.SugaredLogger) (*buildv1alpha1.Build, error) {
-	bs := &build.Spec
-	// Pass service account name from taskrun to build
-	bs.ServiceAccountName = tr.Spec.ServiceAccount
-
+func createRedirectedBuild(kubeclient kubernetes.Interface, ts *v1alpha1.TaskSpec, tr *v1alpha1.TaskRun, cache *entrypoint.Cache, logger *zap.SugaredLogger) (*v1alpha1.TaskSpec, error) {
```

Comment:
another function name which can probably be updated, maybe `createRedirectedTaskRun`?